src/app/pages/vm/dockervm-wizard/dockervm-wizard.component.ts

```diff
@@ -0,0 +1,238 @@
+        {
+          type: 'input',
+          name: 'raw_filename',
+          placeholder : 'filename',
+          tooltip: 'Provide a filename, this file will be created at user specific location ',
+          validation : [ Validators.required ]
+        },
+        {
+          type: 'input',
+          name: 'size',
+          placeholder : 'Define the size (in GiB) for the raw file.',
+          tooltip: 'Type a number of GiB to allocate to the new RAW file.',
+          value: 10,
+          inputType: 'number',
+          min: 10,
+          validation : [ Validators.required ]
+        },
+        {
+          type: 'explorer',
+          name: 'raw_file_directory',
+          placeholder: 'Select a directory',
+          tooltip: 'please select a path for existing directory',
+          explorerType: "directory",
+          initial: '/mnt',
+          validation : [ Validators.required ]
+        },
+        {
+          type: 'input',
+          name: 'sectorsize',
+          placeholder : 'sectorsize',
+          tooltip: '.',
+          value: 0,
+          inputType: 'number',
+          min: 0
+        },
+      ]
+    },
+  ]
+
+  protected releaseField: any;
+  protected currentServerVersion: any;
+  private raw_filename: any;
+  private raw_file_directory: any;
+  private raw_file: any;
+  private nic_attach: any;
+  private nicType:  any;
+  private bootloader: any;
+
+  constructor(protected rest: RestService, protected ws: WebSocketService, 
+    public vmService: VmService, public networkService: NetworkService,
+    protected loader: AppLoaderService, protected dialog: MatDialog, 
+    private router: Router) {
+
+  }
+
+
+  afterInit(entityWizard: EntityWizardComponent) {
+
+    this.networkService.getAllNicChoices().subscribe((res) => {
+      this.nic_attach = _.find(this.wizardConfig[2].fieldConfig, {'name' : 'nic_attach'});
+      res.forEach((item) => {
+        this.nic_attach.options.push({label : item[1], value : item[0]});
+      });
+      ( < FormGroup > entityWizard.formArray.get([2])).controls['nic_attach'].setValue(
+        this.nic_attach.options[0].value
+      )
+
+    });
+    this.ws.call('notifier.choices', [ 'VM_NICTYPES' ]).subscribe((res) => {
+          this.nicType = _.find(this.wizardConfig[2].fieldConfig, {name : "NIC_type"});
+          res.forEach((item) => {
+            this.nicType.options.push({label : item[1], value : item[0]});
+          });
+        ( < FormGroup > entityWizard.formArray.get([2])).controls['NIC_type'].setValue(
+          this.nicType.options[0].value
+        )
+        });
+
+      this.ws.call('vm.random_mac').subscribe((mac_res)=>{
+        ( < FormGroup > entityWizard.formArray.get([2])).controls['NIC_mac'].setValue(mac_res);
+      });
+  
+  }
+  getRndInteger(min, max) {
+    return Math.floor(Math.random() * (max - min + 1) ) + min;
+}
+
+async customSubmit(value) {
+  const path = value.raw_file_directory+ '/' + value.raw_filename+ '_'+ value.name;
```

Comment:
I think here would be safer to do something like:
const path = value.raw_file_directory+ '/' + value.id + '_' + value.name + '_' +value.raw_filename;

Or otherwise users will face out an issue if they try to create two VMs with the same name, vm.name are not unique, but vm.id must be unique.

What you think?